2023-08-30 15:17:00,826 - worker_process - INFO - Worker_process started.
2023-08-30 15:17:01,077 - Cell - INFO - site-1.2674eb88-1fbd-4ba4-8274-36a248f7cba4: created backbone internal connector to tcp://localhost:29726 on parent
2023-08-30 15:17:01,102 - Cell - INFO - site-1.2674eb88-1fbd-4ba4-8274-36a248f7cba4: created backbone external connector to grpc://192.168.100.3:8102
2023-08-30 15:17:01,102 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00001 ACTIVE tcp://localhost:29726] is starting
2023-08-30 15:17:01,102 - nvflare.fuel.f3.sfm.conn_manager - INFO - Connector [CH00002 ACTIVE grpc://192.168.100.3:8102] is starting
2023-08-30 15:17:01,103 - FederatedClient - INFO - Wait for client_runner to be created.
2023-08-30 15:17:01,104 - FederatedClient - INFO - Got client_runner after 0.0006890296936035156 seconds
2023-08-30 15:17:01,104 - FederatedClient - INFO - Got the new primary SP: grpc://192.168.100.3:8102
2023-08-30 15:17:01,114 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: client runner started
2023-08-30 15:17:01,140 - Communicator - INFO - Received from secure_project server  (1114733 Bytes). getTask: train time: 0.01648426055908203 seconds
2023-08-30 15:17:01,141 - FederatedClient - INFO - pull_task completed. Task name:train Status:True 
2023-08-30 15:17:01,141 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: got task assignment: name=train, id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3
2023-08-30 15:17:01,142 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, task_name=train, task_id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3]: invoking task executor LearnerExecutor
2023-08-30 15:17:01,142 - LearnerExecutor - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, task_name=train, task_id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3]: Client trainer got task: train
2023-08-30 15:17:01,142 - MNISTLearner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, task_name=train, task_id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3]: Client site-1 initialized at 
 /home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../2674eb88-1fbd-4ba4-8274-36a248f7cba4/app_site-1 
 with args: Namespace(workspace='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/..', startup='/home/aienode1/NVFlare/sound_cls/mnist-real-world/workspaces/secure_workspace/site-1/startup/../startup', token='d2c82c93-15f6-4e9e-8318-e0e19a0309fe', ssid='ebc6125d-0a56-4688-9b08-355fe9e4d61a', job_id='2674eb88-1fbd-4ba4-8274-36a248f7cba4', client_name='site-1', sp_target='192.168.100.3:8102', parent_url='tcp://localhost:29726', fed_client='fed_client.json', set=['secure_train=true', 'uid=site-1', 'org=nvidia', 'config_folder=config', 'print_conf=True'], local_rank=0, train_config='config/config_train.json', client_config='config/config_fed_client.json', config_folder='config', env='config/environment.json')
2023-08-30 15:17:02,435 - ClientRunner - ERROR - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, task_name=train, task_id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3]: Processing error from executor LearnerExecutor: ValueError: num_samples should be a positive integer value, but got num_samples=0
2023-08-30 15:17:02,435 - ClientRunner - ERROR - Traceback (most recent call last):
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/private/fed/client/client_runner.py", line 254, in _process_task
    reply = executor.execute(task.name, task.data, fl_ctx, self.task_abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 86, in execute
    return self.train(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/nvflare/app_common/executors/learner_executor.py", line 99, in train
    validate_result: Shareable = self.learner.validate(shareable, fl_ctx, abort_signal)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 357, in validate
    self._create_datasets(fl_ctx)
  File "/home/aienode1/NVFlare/sound_cls/pt/learners/sound_learner.py", line 165, in _create_datasets
    self.train_loader = self.train_datamgr.get_loader(self.transform_train, batch_size=self.batch_size, shuffle=True, num_workers=self.num_workers, train=True)
  File "/home/aienode1/NVFlare/sound_cls/pt/utils/data_manager.py", line 36, in get_loader
    loader = data.DataLoader(dataset=dataset, shuffle=shuffle, batch_size=batch_size, num_workers=num_workers, drop_last=True, collate_fn=self.pad_seq)
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 351, in __init__
    sampler = RandomSampler(dataset, generator=generator)  # type: ignore[arg-type]
  File "/home/aienode1/anaconda3/lib/python3.10/site-packages/torch/utils/data/sampler.py", line 107, in __init__
    raise ValueError("num_samples should be a positive integer "
ValueError: num_samples should be a positive integer value, but got num_samples=0

2023-08-30 15:17:02,436 - FederatedClient - INFO - Starting to push execute result.
2023-08-30 15:17:02,440 - Communicator - INFO -  SubmitUpdate size: 575 Bytes. time: 0.0032384395599365234 seconds
2023-08-30 15:17:02,441 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, task_name=train, task_id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3]: result sent to server for task: name=train, id=3c7fe5f6-680a-4d53-a056-c91eabfa6ea3
2023-08-30 15:17:03,311 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4, peer=secure_project, peer_run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: received aux request from Server to end current RUN
2023-08-30 15:17:03,312 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: ABORT (RUN) command received
2023-08-30 15:17:03,312 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: ABORT (RUN) requests end run events sequence
2023-08-30 15:17:03,312 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: ABOUT_TO_END_RUN fired
2023-08-30 15:17:03,312 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: END_RUN fired
2023-08-30 15:17:04,443 - ClientRunner - INFO - [identity=site-1, run=2674eb88-1fbd-4ba4-8274-36a248f7cba4]: run method requests end run events sequence
2023-08-30 15:17:04,457 - FederatedClient - INFO - Shutting down client run: site-1
2023-08-30 15:17:06,433 - MPM - INFO - MPM: Good Bye!
